utilities/src/main/java/org/craftercms/commons/file/stores/S3Utils.java

```diff
@@ -29,32 +31,65 @@
 public class S3Utils {
 
     /**
-     * Creates an {@code AmazonS3} based on the given profile config.
+     * Creates an {@link S3Client} based on the given profile config.
      *
      * @param profile the configuration profile
      *
-     * @return a client to an Amazon S3 account
+     * @return a client to an S3Client account
      */
-    public static final AmazonS3 createClient(AbstractAwsProfile profile, boolean useCustomEndpoint) {
-        AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
-                                                             .withCredentials(profile.getCredentialsProvider());
+    public static final S3Client createClient(AbstractAwsProfile profile, boolean useCustomEndpoint) {
+        S3ClientBuilder builder = S3Client.builder()
```

Comment:
Is it possible to extract the configuring of this builder to one or two generic methods so we can share between `createClient` and `createAsyncClient` methods?
e.g.: `AwsClientBuilder`, `SdkClientBuilder`

Reply:
Updated to use generic function